### PR TITLE
Promote modules/govuk to a deployment

### PR DESCRIPTION
**Pull request by richardTowers**
Previously we had the govuk-test deployment, which was supposed to
configure things which were specific to the test environment and then
pass them down to the govuk module.

But then we ended up pulling (almost) all of the environment specific
configuration out into .tfvars files in variables/test/*.tfvars, so the
govuk-test deployment ended up not knowing very much about the test
environment. Instead, it just became a really annoying layer of
indirection - any variable or outputs needed to be passed through both
the deployment and the module.

Removing this layer of indirection should make our lives much easier.

The terraform plan for this is going to be an absolute abomination,
but I think we should be able to fix that with:

```
terraform state list | while read -ra line; do
  # If the line starts with `module.govuk`
  if grep "^module\.govuk\." <<< "$line"; then
    # Then move it in the statefile to a location without the module.govuk. prefix
    terraform state mv "$line" "${line#module.govuk.}"
  fi
done
```

### CODE REVIEW

File: terraform/deployments/govuk-test/outputs.tf [hunk 1]
```diff
@@ -2,26 +2,6 @@ output "private_subnets" {
...
-  value = module.govuk.signon_security_groups
-}
-
-output "content-store_security_groups" {
-  value = module.govuk.content_store_security_groups
-}
-
-output "draft-content-store_security_groups" {
-  value = module.govuk.draft_content_store_security_groups
-}
```
Comment (bilbof) [inline terraform/deployments/govuk-test/outputs.tf:23]: OK to remove these for now, but we are likely to need them or an equivalent later on.

The reason for these is that unscheduled tasks in ECS must have the same security groups as the ECS Service they would be associated with (e.g. for db migrations, but also for ad-hoc rake tasks like `tools/run-task.sh`).

File: terraform/deployments/govuk-publishing-platform/outputs.tf [hunk 2]
```diff
@@ -2,26 +2,6 @@ output "private_subnets" {
...
-  value = module.govuk.signon_security_groups
-}
-
-output "content-store_security_groups" {
-  value = module.govuk.content_store_security_groups
-}
-
-output "draft-content-store_security_groups" {
-  value = module.govuk.draft_content_store_security_groups
-}
```
Comment (bilbof) [inline terraform/deployments/govuk-publishing-platform/outputs.tf:23]: OK to remove these for now, but we are likely to need them or an equivalent later on.

The reason for these is that unscheduled tasks in ECS must have the same security groups as the ECS Service they would be associated with (e.g. for db migrations, but also for ad-hoc rake tasks like tools/run-task.sh).

**Review by bilbof — APPROVED** [review]
Excellent change 👏

File: terraform/deployments/govuk-publishing-platform/outputs.tf (hunk 2 above)
Comment (richardTowers) [reply terraform/deployments/govuk-publishing-platform/outputs.tf:23]: Ah, I didn't realise that. I'll put them back.

Comment (bilbof) [reply terraform/deployments/govuk-publishing-platform/outputs.tf:23]: That's my bad - sorry, they should've had descriptions on them.

Comment (richardTowers) [reply terraform/deployments/govuk-publishing-platform/outputs.tf:23]: They're back now :)

**Review by bilbof — APPROVED** [review]
Great stuff 🥇

**Pull request merged**